CondCore/PopCon/src/OnlinePopCon.cc

```diff
@@ -0,0 +1,116 @@
+#include "CondCore/CondDB/interface/ConnectionPool.h"
+#include "CondCore/PopCon/interface/OnlinePopCon.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+//#include <iostream>
+
+namespace popcon {
+
+  constexpr const char* const OnlinePopCon::s_version;
+
+  OnlinePopCon::OnlinePopCon(const edm::ParameterSet& pset)
+      : m_targetSession(),
+        m_targetConnectionString(pset.getUntrackedParameter<std::string>("targetDBConnectionString", "")),
+        m_authPath(pset.getUntrackedParameter<std::string>("authenticationPath", "")),
+        m_authSys(pset.getUntrackedParameter<int>("authenticationSystem", 1)),
+        m_recordName(pset.getParameter<std::string>("record")),
+        m_useLockRecors(pset.getUntrackedParameter<bool>("useLockRecords", false)) {
+    edm::LogInfo("OnlinePopCon")
+        << "This is OnlinePopCon (Populator of Condition) v" << s_version << ".\n"
+        << "Please report any problem and feature request through the JIRA project CMSCONDDB.\n";
+  }
+
+  OnlinePopCon::~OnlinePopCon() {
+    if (!m_targetConnectionString.empty()) {
+      m_targetSession.transaction().commit();
+    }
+  }
+
+  cond::persistency::Session OnlinePopCon::preparePopCon() {
+    // Initialization almost identical to PopCon
+    const std::string& connectionStr = m_dbService->session().connectionString();
+    m_dbService->forceInit();  // FIXME: should we run this?
```

Comment:
Do we know anything new about this? I was worried if it's gonna be compatible with the initialization you wrote but it seems like the only thing you do which original PopCon didn't is starting the logger and the `lockRecords()`. It doesn't seem like it would conflict - we can leave it and see if it works when we'll be able to test it.

Reply:
I agree, I'll remove the commented line for now.